### PR TITLE
Shallow L1 functionallty + drawIO export

### PR DESCRIPTION
**Pull request by MrSkooma**
This pull request introduces Shallow L1 functionality and exporting plots to drawIO-compatible formats.

### CODE REVIEW

File: src/probabilistic_model/distributions/uniform.py [hunk 1]
```diff
@@ -119,3 +143,10 @@ def plot(self, **kwargs) -> List:
 
     def __hash__(self):
         return hash((self.variable.name, hash(self.interval)))
+
+
+
+    def all_union_of_mixture_points_with(self, other: Self):
```
Comment (tomsch420) [inline src/probabilistic_model/distributions/uniform.py:149]: docstring

File: src/probabilistic_model/Monte_Carlo_Estimator.py
Comment (tomsch420) [inline src/probabilistic_model/Monte_Carlo_Estimator.py:1]: filename should be snake case

File: src/probabilistic_model/learning/jpt/__pycache__/__init__.cpython-38.pyc
Comment (tomsch420) [inline src/probabilistic_model/learning/jpt/__pycache__/__init__.cpython-38.pyc:1]: remove file

File: src/probabilistic_model/learning/nyga_distribution.py [hunk 2]
```diff
@@ -442,3 +443,97 @@ def from_uniform_mixture(cls, mixture: ProbabilisticCircuit) -> Self:
...
+        y_values = []
+        for weight, subcircuit in self.weighted_subcircuits:
+            interval = subcircuit.interval
+            height = weight * subcircuit.pdf_value()
+            x_values += [interval.lower, interval.upper, None]
+            y_values += [height, height, None]
+        return go.Scatter(x=x_values, y=y_values, mode="lines", name=PDF_TRACE_NAME,
+                          line=dict(color=PDF_TRACE_COLOR))
+
+    def cdf_trace(self) -> go.Scatter:
```
Comment (tomsch420) [inline src/probabilistic_model/learning/nyga_distribution.py:461]: the plotting methods here are not needed anymore

File: src/probabilistic_model/probabilistic_circuit/exporter/draw_io_expoter.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    def __init__(self, model: ProbabilisticCircuit):
+        self.model = model
+
+    def export(self) -> drawio_diagram:
+        diagram = drawio_diagram()
+        diagram.add_diagram("Structure", width=1360, height=1864)
+        for node in self.model.nodes:
+            diagram.add_node(id=str(hash(node)), **node.draw_io_style())
+
+        for source, target in self.model.unweighted_edges:
```
Comment (tomsch420) [inline src/probabilistic_model/probabilistic_circuit/exporter/draw_io_expoter.py:19]: there is a calculation for the positions in pc.plot_structure. Perhaps use that one for positioning

File: src/probabilistic_model/probabilistic_circuit/nx/probabilistic_circuit.py [hunk 4]
```diff
@@ -5,18 +5,30 @@
 from abc import abstractmethod
 import random
 
+import queue
+import random
+from typing import Tuple, Iterable, TYPE_CHECKING
```
Comment (tomsch420) [inline src/probabilistic_model/probabilistic_circuit/nx/probabilistic_circuit.py:10]: typing_extensions

File: src/probabilistic_model/probabilistic_circuit/nx/probabilistic_circuit.py [hunk 5]
```diff
@@ -187,6 +198,60 @@ def marginal(self, *args, **kwargs) -> Optional[Self]:
     def moment(self, *args, **kwargs):
         raise NotImplementedError
 
+    def plot_structure(self):
+        """
+        Plot the structure of the circuit.
+        """
+        # create the subgraph with this node as root
+        subgraph = nx.subgraph(self.probabilistic_circuit, list(nx.descendants(self.probabilistic_circuit, self)) +
```
Comment (tomsch420) [inline src/probabilistic_model/probabilistic_circuit/nx/probabilistic_circuit.py:206]: use the plot_structure from the ProbabilisticCircuit class instead

File: src/probabilistic_model/probabilistic_circuit/nx/probabilistic_circuit.py [hunk 6]
```diff
@@ -187,6 +198,60 @@ def marginal(self, *args, **kwargs) -> Optional[Self]:
...
+        nx.draw_networkx_labels(subgraph, positions, labels)
+
+    def draw_io_style(self) -> Dict[str, Any]:
+        return {
+            "style": self.label,
+            "width": 30,
+            "height": 30,
+            "label": self.representation
+        }
+def nodes_weights(circuit: ProbabilisticCircuit) -> dict:
```
Comment (tomsch420) [inline src/probabilistic_model/probabilistic_circuit/nx/probabilistic_circuit.py:235]: docstring

File: src/probabilistic_model/probabilistic_circuit/nx/probabilistic_circuit.py [hunk 7]
```diff
@@ -1045,6 +1119,13 @@ def unweighted_edges(self):
 
         return unweighted_edges
 
+
+    def plot(self, **kwargs):
```
Comment (tomsch420) [inline src/probabilistic_model/probabilistic_circuit/nx/probabilistic_circuit.py:1123]: these should not have plotting anymore

File: test/__pycache__/test_distributions.cpython-38.pyc
Comment (tomsch420) [inline test/__pycache__/test_distributions.cpython-38.pyc:1]: remove this entire folder

File: test/test_nx/test_exporter.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+        variables = infer_variables_from_dataframe(df, min_likelihood_improvement=1)
+
+        model = JPT(variables, min_samples_leaf=0.9)
+        model.fit(df)
+        cls.model = model
+
+    def test_export_to_drawio(self):
+        diagram = DrawIoExporter(self.model.probabilistic_circuit).export()
+        diagram.dump_file("test.drawio")
```
Comment (tomsch420) [inline test/test_nx/test_exporter.py:41]: use a tempfile here

**Review by tomsch420 — CHANGES_REQUESTED** [review]

File: src/probabilistic_model/probabilistic_circuit/nx/probabilistic_circuit.py (hunk 7 above)
Comment (MrSkooma) [reply src/probabilistic_model/probabilistic_circuit/nx/probabilistic_circuit.py:1123]: i assumed that plotly_layout is not part of it

**Review by tomsch420 — CHANGES_REQUESTED** [review]
Its good. Just resolve the merge conflict and then this is finished.

**Pull request merged**